src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -14,7 +15,114 @@
     from cryptography.hazmat.backends.openssl.backend import Backend
 
 
-class _CipherContext:
+class _CipherContext(metaclass=abc.ABCMeta):
```

Comment:
Why is a new interface required? This seems to match our existing public cipher context ABC.

Reply:
It's an interface for the backend cipher context, which needs a couple of methods not present in the existing public `CipherContext` ABC:

```python
    @abc.abstractmethod
    def authenticate_additional_data(self, data: bytes) -> None:
        ...

    @abc.abstractmethod
    def finalize_with_tag(self, tag: bytes) -> bytes:
        ...
```

I think it also makes sense to have separate interfaces for public vs backend ctx, since the public context wraps the backend one:
```python
    def _wrap_ctx(
        self, ctx: _BackendCipherContext, encrypt: bool
    ) -> typing.Union[
        AEADEncryptionContext, AEADDecryptionContext, CipherContext
    ]:
        if isinstance(self.mode, modes.ModeWithAuthenticationTag):
            if encrypt:
                return _AEADEncryptionContext(ctx)
            else:
                return _AEADDecryptionContext(ctx)
        else:
            return _CipherContext(ctx)
  ```